xilem_web/src/as_template.rs

```diff
@@ -0,0 +1,132 @@
+            ctx.enable_hydration();
+            ctx.push_hydration_node(node);
+            let (mut el, mut state) = prev.build(ctx);
+            el.node.apply_props(&mut el.props);
+            if !is_already_hydrating {
+                ctx.disable_hydration();
+            }
+            let pod_mut = PodMut::new(&mut el.node, &mut el.props, None, false);
+            self.0.rebuild(prev, &mut state, ctx, pod_mut);
+
+            (el, state)
+        } else {
+            let (element, state) = self.0.build(ctx);
+
+            let template: web_sys::Node = element
+                .node
+                .as_ref()
+                .clone_node_with_deep(true)
+                .unwrap_throw();
+
+            ctx.templates.insert(type_id, (template, self.0.clone()));
+            (element, state)
+        };
+        let state = AsTemplateState {
+            view_state,
+            dirty: false,
+        };
+        (element, state)
+    }
+
+    fn rebuild<'el>(
+        &self,
+        prev: &Self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: xilem_core::Mut<'el, Self::Element>,
+    ) -> xilem_core::Mut<'el, Self::Element> {
+        if core::mem::take(&mut view_state.dirty) || !Rc::ptr_eq(&self.0, &prev.0) {
+            self.0
+                .deref()
+                .rebuild(&prev.0, &mut view_state.view_state, ctx, element)
+        } else {
+            // If this is the same value, or no rebuild was forced, there's no need to rebuild
+            element
+        }
+    }
+
+    fn teardown(
+        &self,
+        view_state: &mut Self::ViewState,
+        ctx: &mut ViewCtx,
+        element: xilem_core::Mut<'_, Self::Element>,
+    ) {
+        self.0.teardown(&mut view_state.view_state, ctx, element);
+    }
+
+    fn message(
+        &self,
+        view_state: &mut Self::ViewState,
+        id_path: &[xilem_core::ViewId],
+        message: DynMessage,
+        app_state: &mut State,
+    ) -> xilem_core::MessageResult<Action, DynMessage> {
+        let message_result =
+            self.0
+                .deref()
+                .message(&mut view_state.view_state, id_path, message, app_state);
+        if matches!(message_result, MessageResult::RequestRebuild) {
+            view_state.dirty = true;
+        }
+        message_result
+    }
+}
+
+/// This view creates an internally cached deep-clone of the underlying DOM node.
+/// When the inner view is created again, this will be done more efficiently.
+/// It's recommended to use this as wrapper, when it's expected that the inner `view` will be created a lot, for example in a long list
+///
+/// Additionally it supports memoization when the given `view` is an [`Rc<impl DomView>`].
+///
+/// # Examples
+///
+/// ```
+/// use xilem_web::{as_template, elements::html};
+///
+/// # use xilem_web::interfaces::Element;
+/// # fn component() -> impl Element<()> {
+/// as_template(html::p("This will likely created a lot"))
+/// # }
```

Comment:
Would this help understanding one of the use-cases?
It's some form of instancing, the view could also be called as such (`instanced`)?
```suggestion
/// use xilem_web::{as_template, elements::html, DomFragment};
///
/// fn long_list_fragment() -> impl DomFragment<()> {
///     (0..1000)
///         // Performance increase will be larger with a deeper child views
///         .map(|num| as_template(html::li(format!("number: {num}"))))
///         .collect::<Vec<_>>()
/// }
```

Reply:
@DJMcNab @flosse what are your thoughts on this (see also my other new comment)? (also the name `instanced`?)

Reply:
The example in the doc comments definitely helps :+1: 

I'm not sure about the name :thinking:  
Both `as_tamplate` and `instanced` are fine, but maybe it could be a bit more descriptive?
I wouldn't suggest `cached_deep_clone`, but something more along those words?